src/tanks/Tank.js

```diff
@@ -154,9 +153,6 @@ class Tank extends GameObject {
 		//end of movement stuff
 		
 		//gun stuff
-		const turretDirection = this._getDouble(TURRET_DIRECTION_VAR_NAME);
-		this.mainGun.setTargetGunAngle(turretDirection);
-		this.mainGun.onUpdate();
 		//end of gun stuff
```

Comment:
Yeah, all components have their own update call now (called like 20 lines down or so), so the turret can just use that method instead of being directly told what to do from the tank.

And it is now controlling itself, correct.